src/confluent_kafka/avro/cached_schema_registry_client.py

```diff
@@ -159,10 +173,17 @@ def _send_request(self, url, method='GET', body=None, headers={}):
             raise ClientError("Method {} is invalid; valid methods include {}".format(method, VALID_METHODS))
 
         _headers = {'Accept': ACCEPT_HDR}
+        nbody = body
         if body:
-            _headers["Content-Length"] = str(len(body))
-            _headers["Content-Type"] = "application/vnd.schemaregistry.v1+json"
+            nbody = json.dumps(body).encode('UTF-8')
+            _headers["Content-Length"] = str(len(nbody))
         _headers.update(headers)
+        if url.startswith('https'):
```

Comment:
yes, basically whenever the request is to https client it might have password protected key or not , either way this attribute works while the other only works for https SSL connection with non password protected key.